hugr-core/src/hugr.rs

```diff
@@ -3,6 +3,7 @@
 pub mod hugrmut;
 
 pub(crate) mod ident;
+pub mod internal;
```

Comment:
Leaving it private creates the same behaviour we currently have, where `HugrView`/`HugrMut` can only be implemented locally and the `HugrMutInternals` methods are inaccessible to other crates.

Since this is an _internal_ crate already, we can export the trait and use it in our other crates that require low-level access to the graph.

It may be useful when further splitting up this crate, but the immediate use is accessing `HugrMutInternals::replace_op` for low-level rewriting in `tket2`.